libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -1906,6 +1906,85 @@ MAV_RESULT GCS_MAVLINK::handle_command_long_message(mavlink_command_long_t &pack
     return result;
 }
 
+
+bool GCS_MAVLINK::try_send_mission_message(enum ap_message id)
+{
+    AP_Mission *mission = get_mission();
+    if (mission == nullptr) {
+        return true;
+    }
+    switch (id) {
+    case MSG_CURRENT_WAYPOINT:
+        CHECK_PAYLOAD_SIZE(MISSION_CURRENT);
+        mavlink_msg_mission_current_send(chan, mission->get_current_nav_index());
+        return true;
+    case MSG_MISSION_ITEM_REACHED:
+        CHECK_PAYLOAD_SIZE(MISSION_ITEM_REACHED);
+        mavlink_msg_mission_item_reached_send(chan, mission_item_reached_index);
+        return true;
+    case MSG_NEXT_WAYPOINT:
+        CHECK_PAYLOAD_SIZE(MISSION_REQUEST);
+        queued_waypoint_send();
+        return true;
+    default:
+        return true;
+    }
+    return true;
+}
+
+void NOINLINE GCS_MAVLINK::send_hwstatus()
```

Comment:
Huh.  I was actually wondering whether it was still relevant.

I included it on the basis that it probably couldn't hurt! 

Will remove.